### PR TITLE
Gg/214939 serviceconnection cli auto

### PR DESCRIPTION
**Pull request by ganeshg2491**
- Azure Devops Service endpoint(Service Connection) automation using az cli.

### CODE REVIEW

File: .azuredevops/deploy-bootstrap-serviceconnection.yaml [hunk 1]
```diff
@@ -63,5 +63,14 @@ extends:
                 ScriptArguments: >
                   -SubscriptionName $(subscriptionName) 
                   -KeyVaultName $(ssvPlatformKeyVaultName)
-                  -Tier2ApplicationClientIdSecretName $(tier2ApplicationClientIdSecretName)                   
+                  -Tier2ApplicationClientIdSecretName $(tier2ApplicationClientIdSecretName)    
+              - displayName: Setup AzureRm Service Endpoint
+                Type: AzureCLI
+                scriptPath: 'scripts/Setup-Service-Connection.ps1'
```
Comment (DaRoszko) [inline .azuredevops/deploy-bootstrap-serviceconnection.yaml:69]: I agree we'd need a script for service connection, but why isn't there a file of service connections to setup, or similar, so it's config and declarative driven ? means we can have 1x configuration file of connections and a generic script... @ganeshg2491  @kbabigumira  @ajaysoni-cap

File: .azuredevops/deploy-bootstrap-serviceconnection.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ resources:
       name: DEFRA/adp-pipeline-common
       endpoint: DEFRA
       type: github
-      ref: main 
+      ref: gg/214939-serviceendpointvariables
```
Comment (aadev1) [inline .azuredevops/deploy-bootstrap-serviceconnection.yaml:15]: @ganeshg2491 - Should this be main?

File: scripts/Setup-Service-Connection.ps1
Comment (aadev1) [inline scripts/Setup-Service-Connection.ps1:1]: @ganeshg2491 Maybe call this Initialize-ServiceConnection.ps1 as per approved powershell verbs?

https://learn.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.3

**Comment by ganeshg2491** [comment]
creating another PR with config approach and new naming convestion

**Pull request closed without merging**